### YamlMime:ManagedReference
items:
- uid: System.EntryPointNotFoundException
  id: EntryPointNotFoundException
  children:
  - System.EntryPointNotFoundException.#ctor
  - System.EntryPointNotFoundException.#ctor(System.String)
  - System.EntryPointNotFoundException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.EntryPointNotFoundException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: EntryPointNotFoundException
  nameWithType: EntryPointNotFoundException
  fullName: System.EntryPointNotFoundException
  type: Class
  summary: "A kivétellel, hogy ha egy osztály betöltésére tett kísérlet történt egy belépési metódusa hiánya miatt meghiúsul."
  remarks: "EntryPointNotFoundException kivételt vált ki, ha a közös nyelvi futtatókörnyezet nem tudja betölteni a szerelvényt, mert a szerelvény belépési pont nem tudja azonosítani.  Ez a kivétel is történt, a következő feltételek:-a közös nyelvi futtatókörnyezet nem tudja meghatározni az alkalmazás belépési pont (általában egy `Main` metódus) végrehajtható szerelvényben. Lehet, hogy az alkalmazás belépési pont egy globális vagy `static` metódust, amelynek nincsenek paraméterei, vagy annak csak paraméterként egy karakterlánc-tömbben. A belépési pont visszatérhessen `void`, vagy képes megjeleníteni egy <xref:System.Int32>vagy <xref:System.UInt32>kilépési kód.</xref:System.UInt32> </xref:System.Int32> Egy alkalmazás szerelvény nem határozhat meg egynél több belépési pont.      -Windows dll-fájl egy függvény hívása nem sikerült feloldani, mert a függvény nem található. A következő példában EntryPointNotFoundException kivételt vált ki, mert User32.dll nem tartalmaz olyan funkciót nevű `GetMyNumber`.           [!code-cs[System.EntryPointNotFoundException.Class#1](~/add/codesnippet/csharp/t-system.entrypointnotfo_1.cs)][!code-vb[System.EntryPointNotFoundException.Class#1](~/add/codesnippet/visualbasic/t-system.entrypointnotfo_1.vb) ] -egy függvény hívása a Windows dll-fájl nem lehet feloldani, mert az nevét a metódus hívása nem egyezik meg egy nevet a szerelvényben található.       Gyakran, ennek oka, hogy a <xref:System.Runtime.InteropServices.DllImportAttribute.ExactSpelling?displayProperty=fullName>mező implicit vagy explicit módon beállítása `true`, a hívott metódus egy vagy több karakterlánc paramétereket tartalmaz, és egy ANSI és a Unicode verzió is rendelkezik, és a metódus hívásához használt név nem felel meg a ANSI vagy Unicode verzió neve.</xref:System.Runtime.InteropServices.DllImportAttribute.ExactSpelling?displayProperty=fullName> A következő példa illusztrálja hívása a Windows által `MessageBox` User32.dll függvényt. Mivel az első módszer-definíció <xref:System.Runtime.InteropServices.CharSet?displayProperty=fullName>karakterlánc átadás-előkészítési, az a közös nyelvi keresi a funkció széles karakter verziója `MessageBoxW`, az nevét a metódus hívása helyett `MessageBox`.</xref:System.Runtime.InteropServices.CharSet?displayProperty=fullName>  A második metódusdefiníciót a probléma javításához meghívásával a `MessageBoxW` ahelyett, hogy a `MessageBox` függvény.           [!code-vb[System.EntryPointNotFoundException.Class#2](~/add/codesnippet/visualbasic/t-system.entrypointnotfo_2.vb)][!code-cs[System.EntryPointNotFoundException.Class#2](~/add/codesnippet/csharp/t-system.entrypointnotfo_2.cs) ] -próbál egy dinamikus csatolású függvénytár függvények hívhatók leírással ellátott nevét, hanem az egyszerű nevét.       A C++ fordítóprogram általában állít elő, DLL-funkciók leírással ellátott nevét. Például a következő C++ határozza meg azt a függvény `Double` küldése a TestDll.dll nevű tárban.           [!code-cpp[System.EntryPointNotFoundException.Class#6](~/add/codesnippet/cpp/testdll/testdll.cpp#6) ] , ha a kódot az alábbi példában megpróbálja függvény, EntryPointNotFoundException kivételt vált ki, mert a `Double` függvény nem található.           [!code-cs[System.EntryPointNotFoundException.Class#7](~/add/codesnippet/csharp/t-system.entrypointnotfo_4.cs)][!code-vb[System.EntryPointNotFoundException.Class#7](~/add/codesnippet/visualbasic/t-system.entrypointnotfo_4.vb) ] azonban, ha a függvényt hívták leírással ellátott nevének (ebben az esetben `?Double@@YAHH@Z`), a függvény hívásához szükséges sikeres, az alábbi példában látható módon.                 [!code-cs[System.EntryPointNotFoundException.Class#8](~/add/codesnippet/csharp/t-system.entrypointnotfo_5.cs)][!code-vb[System.EntryPointNotFoundException.Class#8](~/add/codesnippet/visualbasic/t-system.entrypointnotfo_5.vb) ] megtalálhatja például Dumpbin.exe segédprogrammal DLL által exportált funkciók leírással ellátott nevét.            -Kívánt metódus meghívására egy felügyelt szerelvényben, mintha egy nem felügyelt dinamikus csatolású kódtára. Ez a művelet látható, lefordítani a StringUtilities.dll nevű szerelvényt a következő példát.           [!code-cs[System.EntryPointNotFoundException.Class#3](~/add/codesnippet/csharp/t-system.entrypointnotfo_6.cs)][!code-vb[System.EntryPointNotFoundException.Class#3](~/add/codesnippet/visualbasic/t-system.entrypointnotfo_6.vb) ] majd lefordítani és hajtható végre a következő példának, amely megpróbálja felhívni a `StringUtilities.SayGoodMorning` metódust a StringUtilities.dll dinamikus csatolású függvénytár, mintha nem felügyelt kód.       Az eredmény EntryPointNotFoundException kivétel.           [!code-cs[System.EntryPointNotFoundException.Class#4](~/add/codesnippet/csharp/t-system.entrypointnotfo_7.cs)][!code-vb[System.EntryPointNotFoundException.Class#4](~/add/codesnippet/visualbasic/t-system.entrypointnotfo_7.vb) ] elkerülése érdekében a kivételt, adjon hozzá egy hivatkozást, a felügyelt szerelvényt és a hozzáférés a `StringUtilities.SayGoodMorning` metódus csak a felügyelt kódban, más módszerrel szeretné éri el, mint a következő példa.                 [!code-cs[System.EntryPointNotFoundException.Class#5](~/add/codesnippet/csharp/t-system.entrypointnotfo_8.cs)][!code-vb[System.EntryPointNotFoundException.Class#5](~/add/codesnippet/visualbasic/t-system.entrypointnotfo_8.vb) ] -próbál metódus meghívására egy olyan COM DLL-fájl, mintha egy Windows DLL-Fájljában.       Egy COM DLL szeretne használni, válassza ki a **hivatkozás hozzáadása** vegye fel a projektbe egy hivatkozást, és válassza ki a típustárat a Visual Studio beállítás a **COM** fülre.       A kezdeti tulajdonságértékek EntryPointNotFoundException példányt listájáért lásd: a <xref:System.EntryPointNotFoundException.%23ctor%2A>konstruktorok.</xref:System.EntryPointNotFoundException.%23ctor%2A>"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class EntryPointNotFoundException : TypeLoadException
  inheritance:
  - System.TypeLoadException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.EntryPointNotFoundException.#ctor
  id: '#ctor'
  parent: System.EntryPointNotFoundException
  langs:
  - csharp
  name: EntryPointNotFoundException()
  nameWithType: EntryPointNotFoundException.EntryPointNotFoundException()
  fullName: System.EntryPointNotFoundException.EntryPointNotFoundException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Egy új példányát inicializálja a <xref href=&quot;System.EntryPointNotFoundException&quot;> </xref> osztály."
  remarks: "Ez a konstruktor inicializálja a <xref:System.Exception.Message%2A>tulajdonság az új példány egy rendszer által megadott üzenetet, amely leírja, hogy a hiba, például a &quot;Belépési pontja nem található.&quot;</xref:System.Exception.Message%2A> Ez az üzenet figyelembe veszi a rendszer jelenlegi kulturális környezethez.       Az alábbi táblázat a kezdeti tulajdonságértékek <xref:System.EntryPointNotFoundException>.</xref:System.EntryPointNotFoundException> példánya      | Tulajdonság |} Érték |}   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| Egy null hivatkozás (`Nothing` a Visual Basic nyelvben). |}   | <xref:System.TypeLoadException.Message%2A>| Az üzenet honosított hibakarakterlánc. |}</xref:System.TypeLoadException.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public EntryPointNotFoundException ();
    parameters: []
  overload: System.EntryPointNotFoundException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.EntryPointNotFoundException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.EntryPointNotFoundException
  langs:
  - csharp
  name: EntryPointNotFoundException(String)
  nameWithType: EntryPointNotFoundException.EntryPointNotFoundException(String)
  fullName: System.EntryPointNotFoundException.EntryPointNotFoundException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Egy új példányát inicializálja a <xref href=&quot;System.EntryPointNotFoundException&quot;> </xref> osztály egy adott hiba miatt."
  remarks: "A tartalma `message` célja, hogy ember által is értelmezhető. Ez a konstruktor védőfalkezelőbe annak érdekében szükség, hogy ez a karakterlánc a jelenlegi rendszer kulturális környezethez tartozó honosított.       Az alábbi táblázat a kezdeti tulajdonságértékek <xref:System.EntryPointNotFoundException>.</xref:System.EntryPointNotFoundException> példánya      | Tulajdonság |} Érték |}   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| Egy null hivatkozás (`Nothing` a Visual Basic nyelvben). |}   | <xref:System.TypeLoadException.Message%2A>| A üzenetet hiba karakterlánca. |}</xref:System.TypeLoadException.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public EntryPointNotFoundException (string message);
    parameters:
    - id: message
      type: System.String
      description: "A hibaüzenet, amely ismerteti a kivétel az az oka."
  overload: System.EntryPointNotFoundException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.EntryPointNotFoundException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.EntryPointNotFoundException
  langs:
  - csharp
  name: EntryPointNotFoundException(SerializationInfo,StreamingContext)
  nameWithType: EntryPointNotFoundException.EntryPointNotFoundException(SerializationInfo,StreamingContext)
  fullName: System.EntryPointNotFoundException.EntryPointNotFoundException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Egy új példányát inicializálja a <xref href=&quot;System.EntryPointNotFoundException&quot;> </xref> a szerializált adatok osztályt."
  remarks: "Ez a konstruktor nevezik meglévő kölcsönökre az adatfolyam továbbított kivétel objektum deszerializálása közben."
  syntax:
    content: protected EntryPointNotFoundException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "A szerializált objektum adatokat tartalmazó objektum."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "A cél- és környezeti információt."
  overload: System.EntryPointNotFoundException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.EntryPointNotFoundException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.EntryPointNotFoundException
  langs:
  - csharp
  name: EntryPointNotFoundException(String,Exception)
  nameWithType: EntryPointNotFoundException.EntryPointNotFoundException(String,Exception)
  fullName: System.EntryPointNotFoundException.EntryPointNotFoundException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Egy új példányát inicializálja a <xref href=&quot;System.EntryPointNotFoundException&quot;> </xref> megadott hibaüzenet, és hivatkozni kell a belső kivételben találhatók okozza-e ezt a kivételt, a osztályt."
  remarks: "Kivételt vált ki, mert egy előző kivétel közvetlen eredménye tartalmaznia kell egy hivatkozást az előző kivétel az a <xref:System.Exception.InnerException%2A>tulajdonság.</xref:System.Exception.InnerException%2A> A <xref:System.Exception.InnerException%2A>tulajdonság, amely a konstruktorban, vagy egy null hivatkozás át ugyanazon értékét adja vissza (`Nothing` a Visual Basic nyelvben) Ha a <xref:System.Exception.InnerException%2A>tulajdonság nem adja meg a belső kivétel értékét a konstruktor.</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       Az alábbi táblázat a kezdeti tulajdonságértékek <xref:System.EntryPointNotFoundException>.</xref:System.EntryPointNotFoundException> példánya      | Tulajdonság |} Érték |}   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| A belső kivétel referencia. |}   | <xref:System.TypeLoadException.Message%2A>| A üzenetet hiba karakterlánca. |}</xref:System.TypeLoadException.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public EntryPointNotFoundException (string message, Exception inner);
    parameters:
    - id: message
      type: System.String
      description: "A hibaüzenet, amely ismerteti a kivétel az az oka."
    - id: inner
      type: System.Exception
      description: "A kivétellel, hogy a jelenlegi kivétel okozza. Ha a `inner` paraméter nem egy null hivatkozás (<xref uid=&quot;langword_csharp_Nothing&quot; name=&quot;Nothing&quot; href=&quot;&quot;> </xref> a Visual Basic nyelvben), a jelenlegi kivétel a következik be egy <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref> blokk, amely kezeli a belső kivételben találhatók."
  overload: System.EntryPointNotFoundException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.TypeLoadException
  isExternal: false
  name: System.TypeLoadException
- uid: System.EntryPointNotFoundException.#ctor
  parent: System.EntryPointNotFoundException
  isExternal: false
  name: EntryPointNotFoundException()
  nameWithType: EntryPointNotFoundException.EntryPointNotFoundException()
  fullName: System.EntryPointNotFoundException.EntryPointNotFoundException()
- uid: System.EntryPointNotFoundException.#ctor(System.String)
  parent: System.EntryPointNotFoundException
  isExternal: false
  name: EntryPointNotFoundException(String)
  nameWithType: EntryPointNotFoundException.EntryPointNotFoundException(String)
  fullName: System.EntryPointNotFoundException.EntryPointNotFoundException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.EntryPointNotFoundException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.EntryPointNotFoundException
  isExternal: false
  name: EntryPointNotFoundException(SerializationInfo,StreamingContext)
  nameWithType: EntryPointNotFoundException.EntryPointNotFoundException(SerializationInfo,StreamingContext)
  fullName: System.EntryPointNotFoundException.EntryPointNotFoundException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.EntryPointNotFoundException.#ctor(System.String,System.Exception)
  parent: System.EntryPointNotFoundException
  isExternal: false
  name: EntryPointNotFoundException(String,Exception)
  nameWithType: EntryPointNotFoundException.EntryPointNotFoundException(String,Exception)
  fullName: System.EntryPointNotFoundException.EntryPointNotFoundException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.EntryPointNotFoundException.#ctor*
  parent: System.EntryPointNotFoundException
  isExternal: false
  name: EntryPointNotFoundException
  nameWithType: EntryPointNotFoundException.EntryPointNotFoundException
